blockly-core/core/ui/function_editor.js

```diff
@@ -825,7 +825,9 @@ Blockly.FunctionEditor.prototype.setupParametersToolbox_ = function () {
 };
 
 Blockly.FunctionEditor.prototype.addEditorFrame_ = function () {
-  var left = Blockly.hasCategories ?
+  // if Blockly is in readOnly mode, don't pad. Otherwise, pad left
+  // based on the size of either the Toolbox or the Flyout
+  var left = Blockly.readOnly ? 0 : Blockly.hasCategories ?
```

Comment:
Oops, sorry I missed this response! Reads a tiny bit clearer, more importantly the point of truth we care about from this context is "is there a flyout", versus "is this read only [implicitly that means there is not a flyout, so ignore its size]"

Could also imagine a helper e.g. `Blockly.divToMainBlockspaceDistance` that encapsulates this.

We can always change later on if we end up revisiting something related to this